.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+stages:
+  - deploy_to_sonatype
+  - create_key
+
+variables:
+  REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
+
+deploy_to_sonatype:
+  stage: deploy_to_sonatype
+  tags: [ "runner:docker", "size:large" ]
+  image: maven:3.8.2-jdk-8-slim
```

Comment:
This is likely to fail in the future because of DockerHub pull limitations. We should pull from Datadog ECR registry. We don't host this maven image directly so you have the following options:

- Add this maven image to the list of images we mirror, and continue installing awscli on every run. Configure the `deploy_to_sonatype` job to read that maven image.
- Add this maven image to the list of images we mirror, and add a pre-job in this .gitlab-ci file that can build and upload a customized version of the maven image that includes awscli. And configure the `deploy_to_sonatype` job to read the latter image.
- Add a new custom image to our ECR repo, based on maven and that includes awscli. Then configure `deploy_to_sonatype` job to use this image

Reply:
After talking offline, I will add a new custom Docker image to the ECR repo with awscli installed. The custom `Dockerfile` will be included in this PR.